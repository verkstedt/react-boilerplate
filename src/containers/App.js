import React, { Component } from 'react'
import { connect } from 'react-redux'
import { locationActions } from 'actions'
import { func, arrayOf, shape, string } from 'prop-types'
import { locationsSelector } from 'reducers/locations'

import Location from 'components/Location'
import Button from 'components/Button'

export class App extends Component {
  static propTypes = {
    fetchLocation: func.isRequired,
    locations: arrayOf(
      shape({
        name: string.isRequired
      })
    )
  }

  static defaultProps = {
    locations: []
  }

  fetchLocation = () => {
    this.props.fetchLocation({ id: 'Julius-Leber-BrÃ¼cke' })
  }

  renderLocations() {
    const { locations } = this.props
    return locations.map(location =>
      <Location location={location} key={location.name} />
    )
  }

  render() {
    return (
      <div>
        <h1>Hello world!</h1>
        <Button onClick={this.fetchLocation}>
          click to fetch station info
        </Button>
        {this.renderLocations()}
      </div>
    )
  }
}

export default connect(locationsSelector, {
  fetchLocation: locationActions.fetchStart
})(App)
