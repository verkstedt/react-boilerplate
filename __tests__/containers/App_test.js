import React from 'react'
import { shallow, mount } from 'enzyme'
import renderer from 'react-test-renderer'

import { App } from 'containers/App'

describe('<App />', () => {
  it('should dispatch fetchLocation on click', () => {
    const spy = jest.fn()
    const wrapper = mount(<App fetchLocation={spy} />)
    wrapper.find('button').simulate('click')
    expect(spy.mock.calls.length).toEqual(1)
  })

  it('renders locations when passed in', () => {
    const spy = jest.fn()
    const wrapper = mount(
      <App locations={[{ name: 'location 1' }]} fetchLocation={spy} />
    )
    expect(wrapper.contains(<p>location 1</p>)).toBeTruthy()
  })

  it('renders correctly', () => {
    const spy = jest.fn()
    const button = renderer
      .create(<App fetchLocation={spy} />)
      .toJSON()
    expect(button).toMatchSnapshot()
  })
})
