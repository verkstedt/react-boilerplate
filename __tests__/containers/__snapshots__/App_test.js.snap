// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<div>
  <h1>
    Hello world!
  </h1>
  <button
    onClick={[Function]}
  >
    click to fetch station info
  </button>
</div>
`;
